pkg/agent/attestor/node/node.go

```diff
@@ -98,7 +102,30 @@ func (a *attestor) Attest(ctx context.Context) (res *AttestationResult, err erro
 		log.WithField(telemetry.SPIFFEID, svid[0].URIs[0].String()).Info("SVID loaded")
 	}
 
-	return &AttestationResult{Bundle: bundle, SVID: svid, Key: key}, nil
+	return &AttestationResult{Bundle: bundle, SVID: svid, Key: key, Reattestable: reattestable}, nil
+}
+
+func (a *attestor) Attest(ctx context.Context) (res *AttestationResult, err error) {
+	log := a.c.Log
+
+	bundle, err := a.loadBundle()
+	if err != nil {
+		return nil, err
+	}
+
+	_, key, err := a.loadSVID(ctx)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
These changes are in, it looks a lot better now. Still need to fill in unit tests for the new organization.

1. `ServerStream` is just exported for now. Can move to its own file if that works better.
2. Not reusing connection right now. if we reuse connection is there an edge case where the bundle gets rotated inbetween?

Reply:
1. Exported for now seems fine to me.
2. That situation is unlikely under a healthy deployment since SPIRE servers rotate CAs and publish to the bundle well in advance of using the CA (meaning that even if the bundle was updated, the servers would still be using the previous CA). The benefit of re-using a connection is that we avoid a TLS handshake with the server. That seems to mostly be it. I can still go either way. Not sure if other maintainers can see something I'm missing. @amartinezfayo @rturner3 @MarcosDY .

Reply:
Worth implementing to save the tls handshake.